bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/configurable/providers/cli/CliProvider.java

```diff
@@ -0,0 +1,233 @@
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(TypeConverter.stringToDecimal(cliArg.value));
+        } catch (NumberFormatException e) {
+            throw new CliConfigException(cliArg, String.format(INCOMPATIBLE_TYPE_ERROR_MESSAGE, key.variable,
+                                                               key.type, cliArg.value));
+        }
+    }
+
+    @Override
+    public Optional<BString> getAsStringAndMark(Module module, VariableKey key) {
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        return Optional.of(StringUtils.fromString(cliArg.value));
+    }
+
+    @Override
+    public Optional<BArray> getAsArrayAndMark(Module module, VariableKey key) {
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        throw new CliConfigException(cliArg, String.format(CliConstants.CONFIGURATION_NOT_SUPPORTED_FOR_CLI,
+                                                           key.variable, key.type));
+    }
+
+    @Override
+    public Optional<BMap<BString, Object>> getAsRecordAndMark(Module module, VariableKey key) {
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        throw new CliConfigException(cliArg, String.format(CliConstants.CONFIGURATION_NOT_SUPPORTED_FOR_CLI,
+                                                           key.variable, key.type));
+    }
+
+    @Override
+    public Optional<BTable<BString, Object>> getAsTableAndMark(Module module, VariableKey key) {
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        throw new CliConfigException(cliArg, String.format(CliConstants.CONFIGURATION_NOT_SUPPORTED_FOR_CLI,
+                                                           key.variable, key.type));
+    }
+
+    @Override
+    public Optional<BXml> getAsXmlAndMark(Module module, VariableKey key) {
+        Type effectiveType = ((IntersectionType) key.type).getEffectiveType();
+        CliArg cliArg = getCliArg(module, key);
+        if (cliArg.value == null) {
+            return Optional.empty();
+        }
+        try {
+            return Optional.of(TypeConverter.stringToXml(cliArg.value));
+        } catch (BError e) {
+            throw new CliConfigException(cliArg, String.format(INCOMPATIBLE_TYPE_ERROR_MESSAGE, key.variable,
+                                                               effectiveType, cliArg.value));
+        }
+    }
+
+    private CliArg getCliArg(Module module, VariableKey variableKey) {
+        String key = variableKey.variable;
+        String value;
+        if (rootModule.getOrg().equals(module.getOrg())) {
```

Comment:
do we consider the `DEFAULT_MODULE` case? 
For single file execution, how can the user provide args?

Reply:
sinfle bal file case rootModule = module. So this logic will work for single bal files too.